### PR TITLE
ADD REDUCERS

### PR DESCRIPTION
**Pull request by danielsafari143**
In this pull request, I was able to : 
- Setup a Redux store
- Create a slice to save my state and reducers
- Create a directory that will contain all my Redux logic
- Configure a Redux store

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
```
Comment (AmaduKamara) [inline src/redux/books/booksSlice.js]: - [ ] The initialState is expected to have a books array where the books are stored. Kindly do it like this 👍🏽
```js
initialState: {
    books: [],
  }
```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+
+  reducers: {
+    addBook: (state) => state,
+    removeBook: (state) => state,
+  },
```
Comment (AmaduKamara) [inline src/redux/books/booksSlice.js]: - [ ] The addBook action should add a book received from the payload to the books' state. Kindly work on this reducer to properly store books to the books' state 👍🏽
- [ ] Also, the removeBook action is expected to remove a book from the books state using the Array filter() method. Kindly work on that 👍🏽

File: src/redux/categories/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+const initialState = [];
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+
+  reducers: {
+    checkStatus: (state) => state,
+  },
```
Comment (AmaduKamara) [inline src/redux/categories/categoriesSlice.js]: - [ ] The checkStatus action is expected to return a string of texts stating "Under Construction".

```javascript
checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    }
```

File: src/redux/categories/categoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
```
Comment (AmaduKamara) [inline src/redux/categories/categoriesSlice.js]: - [ ] The initialState is expected to have an empty categories array. Kindly do it like this 👍🏽

```javascript
initialState: {
    categories: [],
  },
```

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @danielsafari143,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed 👍🏽
✅ Used redux toolkit as required 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/books/booksSlice.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.filter((item) => item.id !== action.id);
```
Comment (binhussen) [inline src/redux/books/booksSlice.js]: - Kindly `filter()` method does not mutate the original array but instead returns a new array populated with elements of filtered books, so here you are not assigning the returned value to the book state, please assign it to not lose what you removed 👍 
`state.books = state.filter((item) => item.id !== action.id);`

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @danielsafari143 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Highlights

- Used react-redux and toolkit✔️
- Well, Structured Code ✔️
- Used Git flow ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @danielsafari143 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**